aea/cli/scaffold.py

```diff
@@ -39,135 +39,92 @@ def scaffold(ctx: Context):
+        logger.error("A {} with name '{}' already exists. Aborting...".format(item_type, item_name))
         sys.exit(1)
 
     try:
         agent_name = ctx.agent_config.agent_name
-        logger.info("Adding connection scaffold '{}' to the agent '{}'...".format(connection_name, agent_name))
+        logger.info("Adding {} scaffold '{}' to the agent '{}'...".format(item_type, item_name, agent_name))
 
-        Path("connections").mkdir(exist_ok=True)
+        Path(item_type_plural).mkdir(exist_ok=True)
 
         # create the connection folder
-        dest = Path(os.path.join("connections", connection_name))
+        dest = Path(os.path.join(item_type_plural, item_name))
 
         # copy the skill package into the agent's supported skills.
-        src = Path(os.path.join(AEA_DIR, "connections", "scaffold"))
-        logger.debug("Copying connection modules. src={} dst={}".format(src, dest))
+        src = Path(os.path.join(AEA_DIR, item_type_plural, "scaffold"))
+        logger.debug("Copying {} modules. src={} dst={}".format(item_type, src, dest))
 
         shutil.copytree(src, dest)
 
         # add the connection to the configurations.
-        logger.debug("Registering the connection into {}".format(DEFAULT_AEA_CONFIG_FILE))
-        ctx.agent_config.connections.add(connection_name)
+        logger.debug("Registering the {} into {}".format(item_type, DEFAULT_AEA_CONFIG_FILE))
+        existing_item_list.add(item_name)
         ctx.agent_loader.dump(ctx.agent_config, open(os.path.join(ctx.cwd, DEFAULT_AEA_CONFIG_FILE), "w"))
 
+        # ensure the name in the yaml and the name of the folder are the same
+        config_filepath = os.path.join(ctx.cwd, item_type_plural, item_name, default_config_filename)
+        config = loader.load(open(str(config_filepath)))
+        config.name = item_name
+        loader.dump(config, open(config_filepath, "w"))
+
     except FileExistsError:
-        logger.error("A connection with this name already exists. Please choose a different name and try again.")
+        logger.error("A {} with this name already exists. Please choose a different name and try again.".format(item_type))
         sys.exit(1)
     except ValidationError:
         logger.error("Error when validating the skill configuration file.")
-        shutil.rmtree(os.path.join("connections", connection_name), ignore_errors=True)
+        shutil.rmtree(os.path.join(item_type_plural, item_name), ignore_errors=True)
         sys.exit(1)
     except Exception as e:
         logger.exception(e)
-        shutil.rmtree(os.path.join("connections", connection_name), ignore_errors=True)
+        shutil.rmtree(os.path.join(item_type_plural, item_name), ignore_errors=True)
         sys.exit(1)
 
 
+@scaffold.command()
+@click.argument('connection_name', type=str, required=True)
+@pass_ctx
+def connection(ctx: Context, connection_name: str) -> None:
+    """Add a connection scaffolding to the configuration file and agent."""
+    _scaffold_item(
```

Comment:
Minor comment, you pass all of ctx and its associated attributes. You could also drop the second and third to last arguments and use the `getattr` function to retrieve the correct method (https://docs.python.org/3/library/functions.html#getattr, https://stackoverflow.com/questions/3061/calling-a-function-of-a-module-by-using-its-name-a-string) using the `ctx` and `item_name` as an argument to it.

Reply:
yes your approach is much nicer (I didn't realise you could do this). I've changed the code accordingly,